lib/views/pay/pay_view.dart

```diff
@@ -127,8 +129,32 @@ class _PayViewState extends State<PayView> {
                 setState(() {
                   _invoiceController.text = data!.text!;
                 });
-                boltString = data!.text;
-                invoiceActions(data.text!);
+                if (data!.text!.startsWith('ln')) {
+                  /// This is a lightning invoice
+                  boltString = data.text;
+                  invoiceActions(boltString!);
+                } else if (data.text!.startsWith('tb1')) {
+                  /// This is a btc address
+                  btcAddress = data.text;
+                  if (context.mounted) {
+                    Navigator.of(context).push(MaterialPageRoute(
+                        builder: (context) => NumberPad(
+                              provider: widget.provider,
+                              btcAddress: btcAddress,
+                            )));
+                  }
+                } else {
+                  /// Throwing error
+                  if (context.mounted) {
```

Comment:
add this before make the logic of the address, `if !context.mounted` just return